ui/app/components/app/gas-customization/gas-modal-page-container/gas-modal-page-container.component.js

```diff
@@ -123,19 +123,25 @@ export default class GasModalPageContainer extends Component {
       infoRowProps: { newTotalFiat, newTotalEth, sendAmount, transactionFee },
     } = this.props;
 
-    let tabsToRender = [
-      {
-        name: this.context.t('basic'),
-        content: this.renderBasicTabContent(gasPriceButtonGroupProps),
-      },
-      {
-        name: this.context.t('advanced'),
-        content: this.renderAdvancedTabContent(),
-      },
-    ];
-
+    let tabsToRender;
```

Comment:
[Q] This change looks like a good one, as it prevents calling the render method of the basic tab content entirely when hideBasic is true. Was that the intent here? does this serve a purpose in the new error message?

Reply:
Originally, I introduced this change because when the gas fetch fails without any backup estimate, the `basicEstimates` value will be null and calling `renderBasicTabContent` will crash the extension even if hideBasic is true. But later I removed the `Advanced option` from send page and `edit` for gas row from confirmation page in cases of failures `advanced input option` is already present in the pages , currently only the approve-token page will have the `edit ` option, which calls the gas modal. (this realization came to be when I started replying to this message 😓)

Also when the gas fetch fails without backup, we are disabling submission from the corresponding page, in our test dapp the user cannot reach the approve token page without successfully completing create token, if  that is the scenario everywhere we can maybe remove this changes and use the old one.

I think preventing the rendering of basic tab when hideBasic is true might be a better option as it will results in application crashing if in case the basicEstimates is null, unless there is any specific reason to add both the tabs and remove the `basic tab` in case `hideBasic` is true previously.